coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -757,9 +757,6 @@ func (server *Server) FailJob(ctx context.Context, failJob *proto.FailedJob) (*p
 
 	switch jobType := failJob.Type.(type) {
 	case *proto.FailedJob_WorkspaceBuild_:
-		if jobType.WorkspaceBuild.State == nil {
-			break
-		}
```

Comment:
I am not sure what this was guarding against so maybe removing it has some unintended consequences...

Reply:
I added it back but only gated the update behind it since that appears specifically for updating the state and I suppose we might not want to do that when no state was included but we still want the publish.